aptos-move/move-examples/bonding_curve_launchpad/sources/liquidity_pair.move

```diff
@@ -0,0 +1,277 @@
+            old_apt_reserves: old_apt_reserves,
+            new_fa_reserves: fa_updated_reserves,
+            new_apt_reserves: apt_updated_reserves
+        });
+        event::emit(LiquidityPairSwap {
+            is_fa_else_apt: true,
+            gained: (fa_gained as u128),
+            swapper_address: swapper_address
+        });
+
+        if(apt_updated_reserves > APT_LIQUIDITY_THRESHOLD && liquidity_pair.is_enabled){
+            // Disable Bonding Curve Launchpad pair.
+            liquidity_pair.is_enabled = false;
+            liquidity_pair.is_frozen = false;
+            // Offload onto third party, public DEX.
+            router::create_pool_coin<AptosCoin>(fa_object_metadata, false);
+            add_liquidity_coin_entry_transfer_ref<AptosCoin>(transfer_ref, &resource_signer_holder::get_signer(), fa_object_metadata, false, ((apt_updated_reserves - (apt_updated_reserves / 10)) as u64), ((fa_updated_reserves - (fa_updated_reserves / 10)) as u64), 0, 0);
+            // Send liquidity tokens to dead address.
+            let apt_coin_wrapped = coin_wrapper::get_wrapper<AptosCoin>();
+            let liquidity_obj = liquidity_pool::liquidity_pool(apt_coin_wrapped, fa_object_metadata, false);
+            liquidity_pool::transfer(&resource_signer_holder::get_signer(), liquidity_obj, @0xdead, primary_fungible_store::balance(@resource_account, liquidity_obj));
+
+            event::emit(LiquidityPairGraduated {
+                fa_object_metadata: fa_object_metadata,
+                dex_address: @swap
+            });
+        }
+    }
+
+    //---------------------------DEX-helpers---------------------------
+    //* Transfer_ref avoids circular dependency during graduation that the normal transfer would experience
+    //* from the dispatchable withdraw function.
+    fun add_liquidity_coin_entry_transfer_ref<CoinType>(
+        transfer_ref: &fungible_asset::TransferRef,
+        lp: &signer,
+        token_2: Object<Metadata>,
+        is_stable: bool,
+        amount_1_desired: u64,
+        amount_2_desired: u64,
+        amount_1_min: u64,
+        amount_2_min: u64,
+    ) {
+        let token_1 = coin_wrapper::get_wrapper<CoinType>();
+        let (optimal_amount_1, optimal_amount_2, _) = router::optimal_liquidity_amounts(
+            token_1,
+            token_2,
+            is_stable,
+            amount_1_desired,
+            amount_2_desired,
+            amount_1_min,
+            amount_2_min,
+        );
+        let optimal_1 = coin::withdraw<CoinType>(lp, optimal_amount_1);
+        let optimal_2 = primary_fungible_store::withdraw_with_ref(transfer_ref, signer::address_of(lp), optimal_amount_2);
+        router::add_liquidity_coin<CoinType>(lp, optimal_1, optimal_2, is_stable);
+    }
+
+    //---------------------------Views---------------------------
+    #[view]
+    public fun get_amount_out(fa_reserves: u128, apt_reserves: u128, supplied_fa_else_apt: bool, amountIn: u64): (u64, u64, u128, u128) {
```

Comment:
nit: supplied_fa_else_apt is a long and confusing name. Just swap_to_apt should be good?